src/sentry/migrations/0417_backfill_groupedmessage_substatus.py

```diff
@@ -0,0 +1,70 @@
+# Generated by Django 2.2.28 on 2023-04-07 23:43
+
+from django.db import connection, migrations
+from psycopg2.extras import execute_values
+
+from sentry.models import GroupStatus, GroupSubStatus
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+BATCH_SIZE = 100
+
+UPDATE_QUERY = """
+    UPDATE sentry_groupedmessage
+    SET substatus = data.substatus
+    FROM (VALUES %s) as data (id, substatus)
+    WHERE sentry_groupedmessage.id = data.id
+"""
+
+
+def backfill_substatus(apps, schema_editor):
+    Group = apps.get_model("sentry", "Group")
+
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        Group.objects.filter(status__in=(GroupStatus.UNRESOLVED, GroupStatus.IGNORED)).values_list(
+            "id", "status", "substatus"
+        ),
+        result_value_getter=lambda item: item[0],
+    )
+
+    cursor = connection.cursor()
+    batch = []
+
+    for group_id, status, substatus in queryset:
+        if status == GroupStatus.UNRESOLVED and substatus is None:
+            batch.append((group_id, GroupSubStatus.ONGOING))
+        elif status == GroupStatus.IGNORED and substatus is None:
+            batch.append((group_id, GroupSubStatus.UNTIL_ESCALATING))
```

Comment:
Hmm, are all ignored issues ignored until escalating? Are we translating existing ignores over?

Reply:
For the time being, newly ignored issues will just be set to `UNTIL_ESCALATING`. We have some future work to replace `Group.IGNORED` -> `Group.ARCHIVED`. 

When we actually start implementing the logic to do the state transitions for Ignored to Unresolved(Escalating) or Unresolved(Ongoing), we can see about backfilling Ignored issues with the proper values.

Reply:
For context, there will be a follow up to undo this.